lib/honeycomb/propagation/honeycomb.rb

```diff
@@ -53,16 +53,47 @@ def parse_v1(payload)
 
     # Serialize trace headers
     module MarshalTraceContext
-      def to_trace_header
-        context = Base64.urlsafe_encode64(JSON.generate(trace.fields)).strip
-        encoded_dataset = URI.encode_www_form_component(builder.dataset)
-        data_to_propogate = [
+      def to_trace_header(propagation_context: nil)
+        if propagation_context.nil?
+          trace_id = trace.id
+          span_id = id
+          trace_fields = trace.fields
+          dataset = builder.dataset
+        else
+          trace_id, span_id, trace_fields, dataset = propagation_context
+        end
```

Comment:
This might be overkill, but it seemed appropriate to keep backwards compatibility for a while here. If `to_trace_header` is called by the `Propagator` class, it will have a `propagation_context`. But this module is still included in the Span class, and if called directly will fall back to expecting a `self` of a `Span` instance